### DISCUSSION TITLE
Unable to run AudioKitStream kit

### DISCUSSION THREAD

**Discussion by monopol40** [pschatzmann/arduino-audio-tools#75] [Q&A] [answered]
Since some days I tried to run 

https://www.pschatzmann.ch/home/2021/12/29/the-audiokit-and-the-sam-tts-engine/

with unmodified ESP32 Audio Kit V2.2
Monitor shows this sequence:

load:0x40078000,len:10944
load:0x40080400,len:6388
entry 0x400806b4

and then a couple of

Error:   ->Wire.endTransmission: 2
Error:   ->Wire.endTransmission: 2
....

and then

Error:   codec init failed!
Error:   audio_hal_init
Error:   audio_hal handle is null
[D][BluetoothA2DPSink.cpp:123] start(): start

and BluetoothA2DPSink seems to be ok.

Maybe, there is some initialization problem with

  // setup output
  auto cfg = kit.defaultConfig(TX_MODE);
  kit.begin(cfg);

???

The example https://blog.squix.org/2019/08/esp32-esp-eye-browser-based-spectrum-analyzer.html is running with that board.

Any hints would be great. I'm stucked. And I feel, pschatzmann's Libs for ESP32 Audio Kit V2.2 are really great, because it's Arduino and has the funktionallity of esp-adf.

**Comment by pschatzmann** [comment]
I noticed that the AudioKit using ESP32 >= 2.0.0 is not working. 
Can you try with 1.0.6 ?

**Comment by monopol40** [comment]
@pschatzmann 
Thanks for your answer.
Arduino Boardsmanager shows, esp32 by Espressif Systems version 1.06 is installed. Boards included in this package: ESP32 Dev Module (and 2 WEMOS boards). 1.06 is max value to select for install with boards manager. .
I still have installed your libraries:
1. git clone --recursive https://github.com/pschatzmann/ESP32-A2DP.git
2. git clone --recursive https://github.com/pschatzmann/arduino-audio-tools.git
3. git clone --recursive https://github.com/pschatzmann/arduino-audiokit.git
4. 
The monitor says "Error: ->Wire.endTransmission: 2". Obviously some problems with Wire connection. I've no real knowledge about that stuff, just speculation.

https://github.com/pschatzmann/arduino-audiokit/releases 

says there are two releases, it's second (and newest) release has been changed to "Wire Library as default setting" with arduino-audiokit-02.0.zip.
Should I try first release arduino-audiokit-01.0.zip or should I disable Wire Library. occ. in both releases? If yes, how can I do it?
(If that makes sense, I really just speculate). 

Thanks for your time to read my stuff.

**Comment by pschatzmann** [comment]
Which AudioKit board are you using ? 

The settings can be found in the AudioKitSettings.h file where you define which board you are using. (see #define AUDIOKIT_BOARD)
The i2c (wire) is relevant to communicate with the built in ES8388 audio chip. You can switch back to the original implementation by commenting out the #define AUDIOKIT_USE_WIRE line

**Comment by monopol40** [comment]
@pschatzmann
Thanks for your answer.

I use this board

https://www.banggood.com/de/ESP32-Aduio-Kit-WiFi+-bluetooth-Module-ESP32-Serial-to-WiFi-Audio-Development-Board-with-ESP32-A1S-p-1449256.html?rmmds=myorder&cur_warehouse=CN

(The "old" version with ES8388 audio chip).

I will take over the initial values from running example  
https://blog.squix.org/2019/08/esp32-esp-eye-browser-based-spectrum-analyzer.html is running with that board.

to AudioKitSettings.h  and will report later on here the result.

Best regards and many thanks again!

**Comment by pschatzmann** [comment]
I was testing it with the version 2.2 3478 which uses the ES8388. Please note that I do not support old boards which use the AC101 audio chip.
ps. My board uses the i2c pins GPIO_NUM_33 and GPIO_NUM_32 (defined in baord_pins_config.h) - It seems that there are some 2.2 versions which use the GPIO_NUM_18 and GPIO_NUM_23 instead.

**Comment by pschatzmann** [comment, marked as answer]
Just a update: I have added some DRAFT support for the AC101 chip to the https://github.com/pschatzmann/arduino-audiokit project

**Comment by sekt1953** [comment]
I need a simple media server that can play the messages my Home Assistant sends to the media player, can I do this with your code and how

thanks in advance
Svenn-Erik

**Comment by pschatzmann** [comment]
My software is intended to build audio solutions easily. So you would need to do the programming yourself. You could send text messages from Home Assistant via your chosen protocol (e.g. udp) and on the ESP you just translate the received text into audio with one of my TTS projects. You find links of the supported TTS libraries in the readme.md

**Comment by sekt1953** [comment]
I was thinking of audio message from the media player in HA like sending
audio to a chromcast device, but the idea with TTS can bring me some way,
but not play radio controled by HA

Tanks for the fast response


Den søn. 6. mar. 2022 kl. 12.38 skrev Phil Schatzmann <
***@***.***>:

**Comment by pschatzmann** [comment]
If you manage to figure out to send audio data from Media Player via a protocol that the ESP32 supports (e.g. via TCP ip) that would work as well.  
Or you could use the ESP32 just as Bluetooth Speaker…

**Comment by sekt1953** [comment]
Thanks for the tip!

Den søn. 6. mar. 2022 kl. 12.51 skrev Phil Schatzmann <
***@***.***>:

> If you manage to figure out to send audio data from Media Player via a
> protocol that the ESP32 supports (e.g. via TCP ip) that would work as well.
> Or you could use the ESP32 just as Bluetooth Speaker…
>
>
